### DISCUSSION TITLE
Feature request: default cursor to not blink to save power

### DISCUSSION THREAD

**Discussion by sitsofe** [ghostty-org/ghostty#3479] [Feature Requests, Ideas] [answered]
At the moment one needs to set both `shell-integration-features = no-cursor` and `cursor-style-blink = false` to disable cursor blinking (as mentioned in #2812). I would be interested to know if there is appetite for disabling cursor blinking by default. The rationale is that having a blinking cursor causes the GPU to draw more power and causes more CPU wakeups. If outright disabling is a step too far then perhaps some sort of blink timeout would be a reasonable middle ground?

Discussion about changing default cursor blinking behaviour for other systems:
- [Fedora discussion from 2009 discussing changing the default for GTK apps](https://lists.stg.fedorahosted.org/archives/list/devel@lists.stg.fedoraproject.org/thread/2WUWKRJQMNKU6DPVNPEKX6MWBG3543IG/?sort=date)
- [Microsoft engineer suggestion that Windows Terminal should implement a blink timeout to save battery](https://github.com/microsoft/terminal/issues/5318)
- [Sublime Text changed the cursor blink default to saver power](https://forum.sublimetext.com/t/sublime-text-4-update-faq/58001)

**Comment by mitchellh** [comment]
Note that if Ghostty is unfocused or in the background (obscured) then we stop blinking. It may be interesting to add a configuration option to stop blinking after idle for some given amount of time too.

**Comment by 00-kat** [comment, marked as answer]
This discussion should perhaps be moved to #2363.

**Reply by sitsofe** [reply]
@00-kat `#2363` is a huge discussion with lots of disparate requests and I can't say I'm keen to wade into it. Wouldn't it be better to gauge whether there is appetite to implement a change in default blink behaviour in this narrow discussion? If there's not then by all means leave this issue closed because I feel that's clearer than letting it become lost and getting notifications for topics that are unrelated to this...

**Reply by pluiedev** [reply]
We'd rather have one centralized place to gather opinions from a wider audience. It's easier from an organization perspective. Though if you feel uncomfortable then one of us could forward your suggestion there.

**Reply by sitsofe** [reply]
@pluiedev thanks for the reply! I'm more than happy for one of you to forward my original comment to the `#2363` discussion and I'll watch from the sidelines :-)
